httpx/_client.py

```diff
@@ -56,6 +56,18 @@
 KEEPALIVE_EXPIRY = 5.0
 
 
+def check_not_closed(method: typing.Callable) -> typing.Callable:
```

Comment:
I'd prefer for us *not* to use a decorator style.
I'll use inline comments below to walk through the alternative...

Reply:
I've removed the decorator. Simple is better than complex.